pkgs/top-level/all-packages.nix

```diff
@@ -8653,15 +8653,19 @@ with pkgs;
 
   libgsf = callPackage ../development/libraries/libgsf { };
 
-  # glibc provides libiconv so systems with glibc don't need to build libiconv
-  # separately, but we also provide libiconvReal, which will always be a
-  # standalone libiconv, just in case you want it
-  libiconv = if stdenv ? cross then
-    (if stdenv.cross.libc == "glibc" then libcCross
-      else if stdenv.cross.libc == "libSystem" then darwin.libiconv
-      else libiconvReal)
-    else if stdenv.isGlibc then glibcIconv stdenv.cc.libc
-    else if stdenv.isDarwin then darwin.libiconv
+  # GNU libc provides libiconv so systems with glibc don't need to build
+  # libiconv separately. Additionally, Apple forked/repackaged libiconv so we
```

Comment:
Hmm a few things

1. The newish `glibcIconv` is just some copped headers, so it's safe and simplest to depend on libiconv unconditionally.

2. As far as `-l` goes it reminds me of needing to link pthread or not. Maybe something to be done for both. Alternatively, maybe `glibcIconv` can include a dummy static lib or something, so we can unconditionally link it.

3. This PR preserves the existing native behavior, so it would be best to do setup hooks in a follow-up PR.